dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -297,6 +299,7 @@ pub contract DSSCollection: NonFungibleToken {
             )
             DSSCollection.totalSupply = DSSCollection.totalSupply + 1
             self.numMinted = self.numMinted + 1 as UInt64
+            self.metadata[completionAddress] = "REDEEMED"
```

Comment:
This change gonna explode the `metadata` field, which confuses client to consume the `metadata`.

Reply:
We could PREFIX this field with something like `self.metadata["DUPLICATE_CHECK".concat(completationAddress)] = "REDEEMED"`

> @andrewdamelio is it possible to leverage this map to do the check? ` pub var completedCollections: {Address: [CollectionCompletedWith]}`

Issue with this is `completedCollectionGroup` doesn't NEED to be called, so we can't really count on it always being set unless we enforce it in the mint function.

Reply:
it will be cleaner to have a separate struct to store the data, right? the only issue with that is adding a new struct is a breaking change?